src/middlewares/rbac.ts

```diff
@@ -10,7 +10,7 @@ export const userRoles: Role[] = [
 
 export const userCanOperate = (checkedRole: Role, incomingRole: Role): boolean => userRoles.indexOf(incomingRole) >= userRoles.indexOf(checkedRole);
 
-export default (role: Role) => async (req: Request, res: Response, next: NextFunction) => {
+export default (role: Role) => async (req: NextApiRequest, res: NextApiResponse, next: () => void) => {
   if(req.jwt) {
     const user = await prisma.user.findUnique({ where: { id: req.jwt.userId }, select: { role: true } });
```

Comment:
Swap the lookup to find a user email matching `req.jwt.sub` instead. JWT id !== id in our db necessarily (and it shouldnt. these are separate concerns), but the `sub` value (as in subject I think for base JWT claims) will usually hold the user's unique *inputted* identifier for external auth services. This will end up being email/ phone number most of the time (which we are going to record internally anyway)

Reply:
Final comment here is to ultimately stub this (as in *leave it intentionally in a broken state*) and comment out rbac checking for testing purposes (just throw a "`if(process.env.NODE_ENV === "development") next()`" at the top there if you don't want to think about it). This way once we have to actually uncomment this we're forced to handle whatever integration testing we need between our auth provider and the rbac middleware

Reply:
Great job here in general!